Update with Experiment PHP Server SDK remote evaluation docs (#43)
amplitude/amplitude-dev-center: docs/experiment/sdks/php-sdk.md

---

> title: Experiment PHP SDK

Reviewer: ```suggestion
title: Experiment PHP SDK (Beta)
```

description: Official documentation for Amplitude Experiment's server-side PHP SDK implementation.
icon: simple/php
---

Official documentation for Amplitude Experiment's server-side PHP SDK implementation.

![Packagist](https://img.shields.io/packagist/v/amplitude/experiment-php-server.svg)

> !!!info "SDK Resources"

Reviewer: ```suggestion
!!!beta "SDK Resources"
```

     [:material-github: GitHub](https://github.com/amplitude/experiment-php-server) · [:material-code-tags-check: Releases](https://github.com/amplitude/experiment-php-server/releases)

> !!!note

Reviewer: ```suggestion
```

!!!info "Timeout & Retry Configuration"
    Configure the timeout and retry options to best fit your performance requirements.
    ```php
    <?php
    $experiment = new \AmplitudeExperiment\Experiment();
    $config = (new \AmplitudeExperiment\Remote\RemoteEvaluationConfigBuilder())
                ->fetchTimeoutMillis(500)
                ->fetchRetries(1)
                ->fetchRetryBackoffMinMillis(0)
                ->fetchRetryTimeoutMillis(500)
                ->build();
    $client = $experiment->initializeRemote('<DEPLOYMENT_KEY>', $config);
    ```

#### Configuration

You can configure the SDK client on initialization.

???config "Configuration Options"
    | <div class="big-column">Name</div>  | Description | Default Value |
    | --- | --- | --- |
    | `debug` | Enable additional debug logging. | `false` |
    | `serverUrl` | The host to fetch variants from. | `https://api.lab.amplitude.com` |
    | `fetchTimeoutMillis` | The timeout for fetching variants in milliseconds. This timeout applies to the initial request, not subsequent retries. | `10000` |
    | `fetchRetries` | The number of retries to attempt if a request to fetch variants fails. | `8` |

>     | `fetchRetryBackoffMinMillis` | The minimum (initial) backoff after a request to fetch variants fails. This delay scales according to the the `fetchRetryBackoffScalar` setting. | `500` |

Reviewer: ```suggestion
    | `fetchRetryBackoffMinMillis` | The minimum (initial) backoff after a request to fetch variants fails. This delay scales according to the `fetchRetryBackoffScalar` setting. | `500` |
```

    $config = (new \AmplitudeExperiment\Remote\RemoteEvaluationConfigBuilder())
                ->fetchTimeoutMillis(500)
                ->fetchRetries(1)
                ->fetchRetryBackoffMinMillis(0)
                ->fetchRetryTimeoutMillis(500)
                ->build();
    $client = $experiment->initializeRemote('<DEPLOYMENT_KEY>', $config);
    ```

#### Configuration

You can configure the SDK client on initialization.

???config "Configuration Options"
    | <div class="big-column">Name</div>  | Description | Default Value |
    | --- | --- | --- |
    | `debug` | Enable additional debug logging. | `false` |
    | `serverUrl` | The host to fetch variants from. | `https://api.lab.amplitude.com` |
    | `fetchTimeoutMillis` | The timeout for fetching variants in milliseconds. This timeout only applies to the initial request, not subsequent retries | `10000` |
    | `fetchRetries` | The number of retries to attempt if a request to fetch variants fails. | `8` |
    | `fetchRetryBackoffMinMillis` | The minimum (initial) backoff after a request to fetch variants fails. This delay is scaled by the `fetchRetryBackoffScalar` | `500` |

>     | `fetchRetryBackoffMaxMillis` | The maximum backoff between retries. If the scaled backoff becomes greater than the max, the max is used for all subsequent requests | `10000` |

Reviewer 2: ```suggestion
    | `fetchRetryBackoffMaxMillis` | The maximum backoff between retries. If the scaled backoff becomes greater than the maximum, Experiment uses the  maximum for all subsequent requests. | `10000` |
```

    | `fetchRetryBackoffScalar` | Scales the minimum backoff exponentially. | `1.5` |
    | `fetchRetryTimeoutMillis` | The request timeout for retrying variant fetches. | `10000` |

!!!info "EU Data Center"

>     If you're using Amplitude's EU data center, configure the `serverUrl` option on initialization to `https://api.lab.eu.amplitude.com`

Reviewer 2: ```suggestion
    If you use Amplitude's EU data center, set the `serverUrl` option on initialization to `https://api.lab.eu.amplitude.com`
```
